tests/phpunit/integration/Core/Permissions/PermissionsTest.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * Class Google\Site_Kit\Tests\Core\Permissions\PermissionsTest
+ *
+ * @package   Google\Site_Kit\Tests\Core\Permissions
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Permissions;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Tests\Fake_Site_Connection_Trait;
+use Google\Site_Kit\Tests\TestCase;
+
+/**
+ * @group Permissions
+ */
+class PermissionsTest extends TestCase {
+	use Fake_Site_Connection_Trait;
+
+	public function setUp() {
+		parent::setUp();
+
+		// Use a snapshot of hooks from before Site Kit's `init` ran during
+		// the plugin bootstrap. This ensures the local instance of Permissions is tested.
+		$GLOBALS['wp_filter'] = tests_snapshot_hooks( $GLOBALS['wp_filter__pre_sk_init'] );
```

Comment:
Hmm, I see the comment here, but still don't quite follow why we have to do this. Can you elaborate? We don't use the "original filters" in other somewhat similar class tests either.

Reply:
You're right. I got confused and thought there was a reason we couldn't use our usual approach here but it seems like we can. Thanks 😄